src/main/java/jenkins/plugins/logstash/LogstashNotifier.java

```diff
@@ -88,6 +88,11 @@ public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher,
   }
 
   private boolean perform(Run<?, ?> run, TaskListener listener) {
+    // globally enabled and active. No need to send the data another time.
+    if (run.getAction(LogstashMarkerAction.class) != null)
```

Comment:
This effectively makes you unable to have a single-document log generated from any build if global is enabled?

Reply:
yes but why you would want single document when you have it already line by line? Currently I don't distinguish between globally and set via JobProperty. We could also add a force option the Notifier.

Reply:
IMO best thing would be to have options to enable both modes globally and an ability to override this per-project.

Minimally I'm concerned about a situation when you want to enable the global option but you can't because of some per-existing projects rely on sending the single document

Reply:
Enabling globally to send everything at the end could be done in a `RunListener.onFinalized()` or `onCompleted()`with the limitation that it can't fail the build.
`onFinalized()` would have the advantage that it includes the Final Result line, while `onCompleted()` would allow to write something to the log but which would not end up in the log sent to the indexer.

Reply:
With the RunListener we could even get it right for pipeline as the Result is set when it's run. Avoiding all those questions about how to get the result in.

Reply:
What will not be possible for pipeline is to enable globally the ConsoleLogFilter and then disable it in pipeline script. The logstashSend could be disabled in pipeline.

Reply:
`RunListener.onFinalized()` sounds best then if it would allow us to include the result in the payload.

If it's impossible to disable the ConsoleLogFilter let's just document this -- it's still better to have the possibility to disable the other 3 cases

Reply:
@mwinter69 are you going to work on this in near future?

Reply:
Yes, though it is a complex topic and not yet sure how to implement in detail. 
E.g. when you enable globally the notifier behaviour (just a single event at the end with the log), how to disable this in a project. Currently I would prefer the JobProperty for this. What if one specifies an explicit notifier in his job while it is also globally enabled. Which one should be taken and what when he wants to fail the build if upload fails (a runlistener can't modify the result anymore)? One could also enable the notifier behaviour via the JobProperty to replace the wrapper behaviour enabled globally. Should we allow a force of the notifier even when globally we have the wrapper enabled.

If we want to have full flexibility we might confuse people when they want to configure the jobs.

Maybe one could reduce it to the following.
A job specific setting overrides a global setting. You can disable the global setting without enabling a job setting. if a job enables the wrapper and configures the notifier only the wrapper will be used.

Reply:
Not sure if I understand the "You can disable the global setting without enabling a job setting. " part.
Agree that what is enabled on the job should win with the global setting. Also agree to only choose one if both are configured directly on the job. It would be nice to emit a warning in that case.

Reply:
Consider you have enabled it globally but for some reason, you don't want to upload the logs at all for a certain project.

Reply:
OK, but how do you propose to handle this? There is no explicit disable option at the moment (you could set maxLines=1)

Reply:
via the JobProperty. It gets a checkbox to disable a global setting.